### PR TITLE
Implement dashboard side of postmate communication

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR introduces the `usePostmateParent` hook, which performs the postmate handshake and returns the resulting `ChildAPI` once it resolves.

Additionally, this PR introduces a higher-level hook called 'useCheckoutCommunication` which wraps the `ChildAPI` with convenience functions for the messages we have defined.

# Issues

Refs #6135 (won't be done until we start listening on the paywall side)

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/hooks/useCheckoutCommunication.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  closeModal = 'checkout.closeModal',
+  transactionInfo = 'checkout.transactionInfo',
+}
+
+// This is just a convenience hook that wraps the `emit` function
+// provided by the parent around some communication helpers. Consumers
+// are intended to check the `ready` property to know when to start
+// using the hook. If they do not, runtime errors may occur if any of
+// these functions are called before the postmate handshake completes.
+export const useCheckoutCommunication = () => {
```
Comment (julien51) [inline unlock-app/src/hooks/useCheckoutCommunication.ts]: Probably Overkill but would there be a way to "cache" the calls until ready if but ready yet?

Comment (cnasc) [reply unlock-app/src/hooks/useCheckoutCommunication.ts]: We actually do something like that in the main window side of the current paywall: https://github.com/unlock-protocol/unlock/blob/c00cb0655289eec82a5f1f1e2a5e38f7e7a257ad/paywall/src/unlock.js/PostMessageEmitters/CheckoutIframeMessageEmitter.ts#L65

It's doable, but probably a bit more slippery when we're doing it in/with a hook

Comment (cnasc) [reply unlock-app/src/hooks/useCheckoutCommunication.ts]: I think it makes sense as a future enhancement

Comment (julien51) [reply unlock-app/src/hooks/useCheckoutCommunication.ts]: Thanks

**Pull request merged**